codegens/objective-c/lib/index.js

```diff
@@ -0,0 +1,285 @@
+var _ = require('./lodash'),
+  sanitizeOptions = require('./util').sanitizeOptions,
+  sanitize = require('./util').sanitize,
+  self;
+
+/**
+ * Parses Raw data to fetch syntax
+ *
+ * @param {Object} body Raw body data
+ * @param {String} indent Indent
+ * @param {Boolean} trim Trim request body if this is true.
+ */
+function parseRawBody (body, indent, trim) {
+  var bodySnippet = '';
+  bodySnippet += 'NSData *postData = [[NSData alloc] initWithData:[@"' + sanitize(body.toString(), trim) + '" ' +
+  'dataUsingEncoding:NSUTF8StringEncoding]];\n';
+  bodySnippet += '[request setHTTPBody:postData];\n';
+  return bodySnippet;
+}
+
+/**
+ * Parses GraphQL body
+ *
+ * @param {Object} body GraphQL body
+ * @param {String} indent Indent
+ * @param {Boolean} trim Trim request body if this is true.
+ */
+function parseGraphqlBody (body, indent, trim) {
+  var bodySnippet = '',
+    rawBody = JSON.stringify(body);
+  bodySnippet += 'NSData *postData = [[NSData alloc] initWithData:[@"' + sanitize(rawBody, trim) + '" ' +
+  'dataUsingEncoding:NSUTF8StringEncoding]];\n';
+  bodySnippet += '[request setHTTPBody:postData];\n';
+  return bodySnippet;
+}
+
+/**
+ * Parses URLEncoded body from request to fetch syntax
+ *
+ * @param {Object} body URLEncoded Body
+ * @param {String} indent Indent
+ * @param {Boolean} trim Trim request body is this true.
+ */
+function parseURLEncodedBody (body, indent, trim) {
+  let bodySnippet = '',
+    key,
+    value,
+    first = true;
+  _.forEach(body, function (data) {
+    if (!data.disabled) {
+      key = trim ? data.key.trim() : data.key;
+      value = trim ? data.value.trim() : data.value;
+      if (first) {
+        bodySnippet += 'NSMutableData *postData = [[NSMutableData alloc] initWithData:[@"' +
+        key + '=' + sanitize(value, trim) + '" dataUsingEncoding:NSUTF8StringEncoding]];\n';
+      }
+      else {
+        bodySnippet += '[postData appendData:[@"&' + key + '=' + sanitize(value, trim) +
+        '" dataUsingEncoding:NSUTF8StringEncoding]];\n';
+      }
+      first = false;
+    }
+  });
+  bodySnippet += '[request setHTTPBody:postData];\n';
+  return bodySnippet;
+}
+
+/**
+ * Parses URLEncoded body from request to fetch syntax
+ *
+ * @param {Object} body URLEncoded Body
+ * @param {String} indent Indent
+ * @param {Boolean} trim Trim request body is this true.
+ */
+function parseFormData (body, indent, trim) {
+  let bodySnippet = '',
+    key,
+    value,
+    first = true;
+
+  _.forEach(body, function (data) {
+    key = trim ? data.key.trim() : data.key;
+    value = trim ? data.value.trim() : data.value;
+    if (!data.disabled) {
+      if (first) {
+        bodySnippet += 'NSArray *parameters = @[';
+      }
+      if (!first) {
```

Comment:
Can we use `array.join(',')` syntax? Store all the individual params in an array and then at last use .join()?